Marlin/SanityCheck.h

```diff
@@ -29,6 +29,16 @@
 #define SANITYCHECK_H
 
 /**
+ * Due to the high number of issues related with old versions of Arduino IDE
+ * we are now warning our users to update their toolkits. In a future Marlin
+ * release we will stop supporting old IDE versions and will require user
+ * action to proceed with compilation in such environments.
+ */
+#if defined(ARDUINO) && ARDUINO < 10500
```

Comment:
My comment is only because I don't _know_ whether `ARDUINO` has always been defined by the Arduino library headers. My expectation is that it has, because a version value is always handy. My emphasis in checking that it doesn't exist is only to give allowance for "very old Arduino libs" which fail to define it. I don't think that we need to throw a message like "Marlin requires the Arduino libraries to compile" because if those libs are missing it will be quite obvious.